### PR TITLE
DF-457:Alt tag amend

### PR DESCRIPTION
**Pull request by TNA-Allan**
Related ticket(s):
https://national-archives.atlassian.net/browse/DF-457

## About these changes

- Fixed missing divs
- Removed problematic alt value, added alt without value to grid template
- Restore and Update condition for teaser image
- To look into further to refactor constants using bucket keys : created DF-470 (as mentioned in the DF-457 )to address this bit.

## How to check these changes
ALT value should not appear for any search results.

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly myself before handing over to reviewer.
- [x] Included the ticket number in the PR title to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [x] Added/updated tests and documentation where relevant.

## For Reviewer

Once PR is merged, check and arrange to deploy on platform-sh.

### CODE REVIEW

**Comment by TNA-Allan** [comment]
@mozaki1 / @jamesbiggs - Please can you review this PR. Thanks.

**Comment by TNA-Allan** [comment]
@JohnHeeryTNA , Please can you review FE side of this PR. Thanks.

File: etna/search/templatetags/search_tags.py [hunk 1]
```diff
@@ -218,3 +218,12 @@ def search_title(search_tab) -> str:
...
     return label
+
+
+@register.simple_tag
+def extended_in_operator(lhs_operand, *rhs_operand_list) -> bool:
+    """
+    Input params are template tags
+    Returns True when rhs_operand_list contains lhs_operand value, False otherwise
+    """
+    return (lhs_operand in rhs_operand_list) or False
```
Comment (mozaki1) [inline etna/search/templatetags/search_tags.py:229]: This is not so readable for me unless I read the comments, I suggest by default the function should return true , 
something like that 
if ((lhs_operand not in rhs_operand_list)
return false
return true

File: etna/ciim/constants.py [hunk 2]
```diff
@@ -99,8 +101,8 @@ def __iter__(self):
     [
         Bucket(key="blog", label="Blog posts"),
         Bucket(key="researchGuide", label="Research Guides"),
-        Bucket(key="insight", label="Insights"),
-        Bucket(key="highlight", label="Highlights"),
+        Bucket(key=BucketKeys.INSIGHT.value, label="Insights"),
+        Bucket(key=BucketKeys.HIGHLIGHT.value, label="Highlights"),
         Bucket(key="audio", label="Audio"),
```
Comment (mozaki1) [inline etna/ciim/constants.py:106]: Hi Allan, maybe we should include those hard coded to the enum, this looks slightly half-done; blog,...

Comment (TNA-Allan) [reply etna/ciim/constants.py:106]: Thanks Mo!, Its not half done, the changes here are related to this ticket, since scope of changing constants can cover templates as well I mentioned 
> To look into further to refactor constants using bucket keys

and created DF-470 (as mentioned in the DF-457 )to address this bit.

Comment (mozaki1) [reply etna/ciim/constants.py:106]: Hi Allan, if we are here, we should it, just to be consistent if reviewed by anybody.

File: etna/search/templatetags/search_tags.py (hunk 1 above)
Comment (TNA-Allan) [reply etna/search/templatetags/search_tags.py:229]: Its **native** to Python and a way we can write **powerful and elegant Python one-liners,**
the evaluation is either (True or False) - short-circuit evaluation for OR here is True - **Advantage here over other constructs**.
or (False or False)

**Review by JohnHeeryTNA — COMMENTED** [review]
Hi @TNA-Allan , I think we can completely remove the alt tags on these so the code changes would be:

templates/search/blocks/search_results__list-card--grid.html
line 32
`
`
templates/search/blocks/search_results__list-card.html
line 27
``

**Comment by TNA-Allan** [comment]
> Hi @TNA-Allan , I think we can completely remove the alt tags on these so the code changes would be:
> 
> templates/search/blocks/search_results__list-card--grid.html line 32 ` ` templates/search/blocks/search_results__list-card.html line 27 ``

@sdwilkes : I understand that currently using description is not an accessible value. Will this be replaced with suitable one when available?  Under this scenario, I thought alt tag is a necessary element. Please can you confirm for this change - removal of the alt tag element or removal of alt tag value?

File: etna/ciim/constants.py (hunk 2 above)
Comment (TNA-Allan) [reply etna/ciim/constants.py:106]: > Hi Allan, if we are here, we should it, just to be consistent if reviewed by anybody.

Not sure what you mean.  I've resolved this comment after responding.

File: etna/search/templatetags/search_tags.py (hunk 1 above)
Comment (mozaki1) [reply etna/search/templatetags/search_tags.py:229]: The code you have written will loop through the list and if it finds the element then it uses the OR operator  True OR False ==True  return True or if not then False OR False==Fasle , in fact, you are adding further overhead, Why do you need to do this ?
Just make it simple:
if ((lhs_operand not in rhs_operand_list)
return false

File: etna/ciim/constants.py (hunk 2 above)
Comment (mozaki1) [reply etna/ciim/constants.py:106]: What I mean we do not need to open another ticket just to resolve something we can fix now since we are here.

Comment (TNA-Allan) [reply etna/ciim/constants.py:106]: Its best practice to keep separate. see if you recollect discussion [PR](https://github.com/nationalarchives/ds-wagtail/pull/449)  `{% if form.group.value != 'nonTna' %}` 
The scope is wider `.py, .html` files, so I discussed with PO, team meeting with this consideration went for another ticket. Hope this will be fine with you.

Comment (mozaki1) [reply etna/ciim/constants.py:106]: In this case can you please create a ticket and assign it to yourself, kindly post the ticket number here.

Comment (TNA-Allan) [reply etna/ciim/constants.py:106]: Already mentioned it see above 
> created DF-470 (as mentioned in the DF-457 )to address this bit.

File: etna/search/templatetags/search_tags.py (hunk 1 above)
Comment (TNA-Allan) [reply etna/search/templatetags/search_tags.py:229]: > I think code performance and simplicity are vital, could you please add unit tests, considering all possible outcomes, particularly edge cases like None, empty lists, and exceptions,...

Its easier to understand responses when its in context. I think  your mention _code performance and simplicity are vital_, has nothing to do with this bit, please see comments above on _short-circuit evaluation_ and _pythonic ways_. Why are you mentioning it in this context?

> I have noticed there are fewer unit tests recently

Again, Its easier to understand responses when its in context. Why are you mentioning the above in this context?

> I also encourage TTD as an approach. TDD is common for all programming languages :)

Good to discuss in dev catch-up.  Can you please give two examples where you have deployed TDD in this code base?

> could you please add unit tests, considering all possible outcomes, particularly edge cases like None, empty lists, and exceptions,...

I've added test cases.

Comment (TNA-Allan) [reply etna/search/templatetags/search_tags.py:229]: Pythonic ways - in a sense does show simplicity and also handles performance bit. We can discuss this in dev catch-up along with TDD. Hope this will be fine.

File: etna/search/tests/test_search_tags.py [hunk 3]
```diff
@@ -1,6 +1,10 @@
...
 
-from ..templatetags.search_tags import query_string_exclude, query_string_include
+from ..templatetags.search_tags import (
+    extended_in_operator,
+    query_string_exclude,
+    query_string_include,
+)
 
 
 class QueryStringTest(SimpleTestCase):
```
Comment (mozaki1) [inline etna/search/tests/test_search_tags.py:10]: Can we have a bit more descriptive title, like what the test is actually testing?

File: etna/search/templatetags/search_tags.py (hunk 1 above)
Comment (TNA-Allan) [reply etna/search/templatetags/search_tags.py:229]: Marking as resolved, if unresolved, please respond to above queries to your responses.

File: etna/search/tests/test_search_tags.py (hunk 3 above)
Comment (TNA-Allan) [reply etna/search/tests/test_search_tags.py:10]: This comment is seemed to be linked with the one below - ref `Like the above note please` - please see response in that location.

File: etna/search/tests/test_search_tags.py [hunk 4]
```diff
@@ -57,3 +61,32 @@ def test_non_existant_parameter(self):
         result = query_string_exclude(context, "page", "1")
 
         self.assertEqual(result, "test=true")
+
+
+class TestExtendedInOperator(SimpleTestCase):
```
[the remark this answers is not in the document]
Comment (TNA-Allan) [reply etna/search/tests/test_search_tags.py:66]: It is not clear what you meant.

Assuming you meant you meant `class QueryStringTest(SimpleTestCase)` v/s `class TestExtendedInOperator(SimpleTestCase)` v/s `the pattern is if_outever_then_true,...`

1. These are class names, they have to be camel case, it cannot be undersore as you have mentioned.
2. The `class TestExtendedInOperator(SimpleTestCase)` tests the method `extended_in_operator` therefore the name.
3. Some class test names begin with Test or end with Test, below examples. Currently there is no PEP8 style guide for a Testcase, neither in our code base or standards.
Example:
etna\collections\tests\test_models.py
`class TestResultsPage(TestCase):`
etna\collections\tests\test_models.py
`class TestTopicExplorerIndexPages(TestCase):`
etna\records\tests\test_templatetags.py
`class TestRecordURLTag(SimpleTestCase):`
etna\search\tests\test_views.py
`class SelectedFiltersTest(SimpleTestCase):`

I've added a backlog ticket https://national-archives.atlassian.net/browse/DF-475, we can discuss and refactor this at one go.

No change here.

**Review by jamesbiggs — APPROVED** [review]
Looks good Allan, I liked this test case 
https://github.com/nationalarchives/ds-wagtail/pull/540/files#diff-2c40ae6a8f7c43f23f2e93c7b5f5a7aeeafec6bee425e990e991b42da5facbf4R67
Nice way of assigning the variables etc for the test cases 😃
